Documentation/Yarn-Spec.md

```diff
@@ -0,0 +1,636 @@
+The order of header tags is unspecified.
+
+#### Title Tag
+
+The *title tag* is a specific header tag that uniquely identifies the node.
+The tag name for the title tag must be `Title`.
+The tag text for the title tag must be unique within the file.
+The tag text for the title tag should be unique within the project.
+
+The behaviour of the program when a title tag's text is not unique across the project is unspecified.
+The program could either flag this as an error and abort, choose which node will be the canonical one with that title, or some other form of name conflict resolution.
+
+### Body
+
+A body is comprised of multiple statements.
+A *statement* is a line that is one of the following:
+
+- dialogue
+- commands
+- options
+
+A statement may have optional hashtags.
+A body must have at least one statement.
+
+The body ends when encountering a line that contains the *body delimter* `===`.
+The body delimiter ends both the current node and the body of that node.
+The end of file must not be used in place of a body delimiter.
+
+#### Hashtags
+
+*Hashtags* are metadata associated with the statement they are a part of.
+Hashtags must go at the end of the statement.
+The other components of the statement must end at the hashtag, the hashtag operates effectively as the newline terminator for the statement.
+
+A hashtag starts with the `#` symbol and contain any text up to the newline or another hashtag.
+`#lineID:a10be2` is an example of a hashtag.
+Multiple hashtags must be separated by the `,` symbol.
+`#lineID:a10be2, #return` is an example of multiple hashtags on a line.
+
+### Dialogue Statement
+
+A dialogue statement is a statement that represents a single line of text in the yarn story.
+In most cases dialogue will be the bulk of the content of a node's body.
+Dialogue lines can be interpolated dialogue or raw dialogue.
+
+An *interpolated dialogue* is one where there are in-line expressions in the line.
+Expressions are encapsulated within the `{` and `}` symbols and it is the presence of these symbols that determine if a line is an interpolated one or not.
+The expression inside the `{}` symbols must be a valid expression.
+The result of the expression must be inserted into the dialogue.
+Other than replacing expressions dialogue statments must not be modified by the implementing program, and provided to the game as written.
+
+A *raw dialogue* is a dialogue statement where there are no expressions.
+
+A dialogue statement can contain any characters except for the `#` character.
+
+`{$name}, you are a bold one.` is an example of an interpolated dialogue.
+`General Kenobi, you are a bold one.` is an example of a raw dialogue.
+
+When resolving ambiguity of statements inside the body the dialogue statement must be considered the lowest priority by the implementing program.
+For example `<<Fred Move Left>>` could be read as a command or a dialogue statement, it is to be considered a command by the program.
```

Comment:
??? Not sure what these two lines are trying to say

Reply:
It is saying that the line `<<fred move left>>` is both arguably a command and just a line of normal dialogue.
Because of this an implementing program has to assume it is a command and not a line of dialogue.